### ISSUE TITLE
cinnamon: Crash sometimes when deleting workspaces from the workspace switcher applet

### ISSUE THREAD

**Issue by ooglaboogla** [linuxmint/mint21-beta#89]
Hello. Been following the dev release of Mint 21 for a while before the beta was released. There's an issue that has been there since Cinnamon 5.4 first came along. Expo does not work. The same problem occurs on Arch Linux, since 5.4 became available there. This would have been easy to miss in development, since Expo isn't really set up on the live iso or base install. For years, I have always added the Workspace Switcher Applet to the panel on Cinnamon, and managed the number of workspaces by right-clicking on the switcher and choosing Manage Workspaces (Expo). If you add the Expo Applet to the panel instead, the same thing happens when you try to open it. The screen dims to the wallpaper, but Expo doesn't open. The only way I've found out of the dimmed screen is to Ctrl-Alt-Backspace, and then log back in. Furthermore, if you use the right-click link from the workspace switcher to Add A New Workspace, that always seems to work, but using the Remove The Current Workspace link causes Cinnamon to freeze a few seconds, then go into fallback mode. Not every time, but always on the second time you do it. Like I said, this is a general Cinnamon issue, because it happens on Mint 21 as well as Arch Linux. I installed and updated the Mint 21 Beta iso today to make sure it still occurs, and it does. Expo is still working fine on Mint 20.3 running Cinnamon 5.2.7, as well as on Xubuntu 22.04 running Cinnamon 5.2.7 over XFCE.

**Comment by mtwebster** [comment]
Does the first issue you mention only happen if you don't have any windows open? Or any time?

If so, I think it would be a duplicate of https://github.com/linuxmint/mint21-beta/issues/73.

The other issue I can't reproduce (the crash) - does this only happen with no windows also?

**Comment by ooglaboogla** [comment]
Just checked that. It only happens with no windows open, so somebody may have beat me to it. When I looked through the list of bug reports before starting, there wasn't a #73 yet. I guess reports are coming in fast. The second part about the fallback still happens with windows open.

**Comment by JosephMcc** [comment]
I can reproduce your crash removing workspaces with the workspaces applet. That one doesn't seem to depend on having any windows open.

**Comment by ooglaboogla** [comment]
Also just tested this again on ArcoLinuxB Cinnamon running Cinnamon 5.4. Exact same behavior with Expo not showing up when no windows are open, and still going into fallback most of the time when clicking Remove The Current Workspace, even with windows open on all workspaces.
